### ISSUE TITLE
Delete columns sometimes doesn't work anymore

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#8413]
I am using the merged version of R-Instat

@Vitalis95 and @lilyclements I am pretty sure this bug has followed the merging of pull request #8398.

I read in the usual survey and immediately tried to delete the columns as shown below.  It doesn't work anymore:
It should be easy to fix as I explain below:

![image](https://github.com/africanmathsinitiative/R-Instat/assets/11226469/8dd835fd-eeda-43c9-b247-23d9453ab46b)
.
I suggest the problem is that deleting a column, now checks and deletes the corresponding key columns.  This data frame doesn't have a key.  If I add a key, then the delete works fine.  The code needs to cope with the situation where there is no key.

**Comment by lilyclements** [comment]
Ah yes, a very good point, and quite a big bug to introduce too! My mistake, I have fixed it here #8416 @rdstern

**Comment by rachelkg** [comment]
@lilyclements and @rdstern 
Hello I am having this same issue when using what i think is the most updated version through visual studio. It happens when i try to delete or rename a column.
If this has been fixed in a version that i am not using my apologies.
![2023-07-07_14-25-30](https://github.com/africanmathsinitiative/R-Instat/assets/85707390/06bdffee-b11e-4890-b233-1f20b6612270)
![2023-07-07_14-25-51](https://github.com/africanmathsinitiative/R-Instat/assets/85707390/56fe6a32-9d6b-4432-842b-602b3c4957e9)

**Comment by lilyclements** [comment]
@rachelkg yes I noticed this too, this should all be fixed in #8422 and so should be sorted in the most recent version. Can you check and confirm this for you? If it is ok now, I will close this PR :)

**Comment by rachelkg** [comment]
@lilyclements I can now delete columns successfully but when i try to rename I get this error message. 
![2023-07-11_10-07-39](https://github.com/africanmathsinitiative/R-Instat/assets/85707390/f6f760bd-e625-4624-985d-c50444a4308d)

**Comment by lilyclements** [comment]
@rachelkg a very good point, sorry about that. I've now fixed it in #8433
